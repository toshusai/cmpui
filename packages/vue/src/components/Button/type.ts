export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
