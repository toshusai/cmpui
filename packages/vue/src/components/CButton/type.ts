export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
