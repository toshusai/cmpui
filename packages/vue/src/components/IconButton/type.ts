export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
