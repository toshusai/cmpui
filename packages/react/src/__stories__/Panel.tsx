export const Panel = (
  props: {
    color: string;
    name: string;
  } & React.HTMLAttributes<HTMLDivElement>,
) => {
  return (
    <div
      style={{
        background: props.color,
        width: "100%",
        height: "100%",
        display: "flex",
        overflow: "auto",
        position: "relative",
      }}
      {...props}
    >
      <div
        style={{
          width: "100%",
          pointerEvents: "none",
          height: "100%",
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          color: "white",
          fontSize: "32px",
          position: "absolute",
        }}
      >
        {props.name}
      </div>
      {/* <pre
        style={{
          fontSize: "12px",
          fontFamily: "monospace",
          margin: 0,
          padding: 8,
        }}
      >
        <code>{JSON.stringify(node?.node.toJson(), null, 4)}</code>
      </pre> */}
    </div>
  );
};
