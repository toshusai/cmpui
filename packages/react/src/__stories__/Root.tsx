export const Root = (props: { children: React.ReactNode }) => {
  return (
    <div
      style={{
        height: "512px",
        width: "100vw",
        display: "flex",
        padding: "4px",
        boxSizing: "border-box",
      }}
    >
      {props.children}
    </div>
  );
};
