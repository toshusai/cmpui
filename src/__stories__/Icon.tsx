export function Icon({
  name,
  filled,
  style,
  size,
  ...props
}: {
  name: string;
  filled?: boolean;
  size?: number;
} & React.HTMLAttributes<HTMLSpanElement>) {
  return (
    <span
      className="material-symbols-outlined"
      data-filled={filled}
      {...props}
      style={{
        fontSize: size,
        ...style,
      }}
    >
      {name}
    </span>
  );
}
