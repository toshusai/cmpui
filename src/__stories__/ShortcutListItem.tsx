import { ListItem } from "..";
import "./stories.css";

export function ShortcutListItem({
  disabled,
  shortcut,
  as = ListItem,
  children,
  ...props
}: {
  children: React.ReactNode;
  shortcut?: string;
  disabled?: boolean;
  as?: React.ElementType;
} & React.ComponentProps<typeof ListItem>) {
  const Component = as;
  return (
    <Component disabled={disabled} className="shortcut-list-item" {...props}>
      <div style={{ width: "100%" }}>{children}</div>
      {shortcut && (
        <div
          style={{
            fontSize: "10px",
            marginLeft: 16,
            whiteSpace: "nowrap",
          }}
        >
          {shortcut}
        </div>
      )}
    </Component>
  );
}
