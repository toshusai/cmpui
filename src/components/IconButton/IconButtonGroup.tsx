import React, { forwardRef } from "react";
import { classNames } from "../../utils/classNames";

export const IconButtonGroup = forwardRef<
  HTMLDivElement,
  {
    children: React.ReactNode;
    className?: string;
  }
>((props, ref) => {
  return (
    <div
      ref={ref}
      className={classNames("cmpui_icon-button__group", props.className)}
    >
      {props.children}
    </div>
  );
});
