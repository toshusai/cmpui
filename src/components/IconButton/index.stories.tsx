import type { Meta, StoryObj } from "@storybook/react";
import { IconButton } from ".";
import { Icon } from "../../__stories__/Icon";
import { IconButtonGroup } from "./IconButtonGroup";

const meta: Meta<typeof IconButton> = {
  title: "Components/Actions/IconButton",
  component: IconButton,
};

export default meta;

type Story = StoryObj<typeof IconButton>;

const corner = (
  <svg
    width="6"
    height="6"
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    style={{
      position: "absolute",
      bottom: 4,
      right: 4,
      transform: "scaleX(-1)",
    }}
  >
    <title>corner</title>
    <path d="M0 24V0.00427246L24 24H0Z" fill="currentColor" />
  </svg>
);

export const Basic: Story = {
  render: () => {
    return (
      <div style={{ display: "flex", gap: 4 }}>
        <IconButton>
          <Icon filled name="arrow_selector_tool" style={{ fontSize: 20 }} />
        </IconButton>
        <IconButton selected>
          <Icon name="square" style={{ fontSize: 20 }} />
          {corner}
        </IconButton>
        <IconButton>
          <Icon name="title" style={{ fontSize: 20 }} />
        </IconButton>
      </div>
    );
  },
};

export const Selected: Story = {
  render: () => {
    return (
      <IconButton selected>
        <Icon name="star" style={{ fontSize: 20 }} />
      </IconButton>
    );
  },
};

export const Disabled: Story = {
  render: () => {
    return (
      <IconButton disabled>
        <Icon name="star" style={{ fontSize: 20 }} />
      </IconButton>
    );
  },
};

export const Small: Story = {
  render: () => {
    return (
      <IconButton size="S">
        <Icon name="star" style={{ fontSize: 16 }} />
      </IconButton>
    );
  },
};

export const Group: Story = {
  render: () => {
    return (
      <IconButtonGroup>
        <IconButton>
          <Icon filled name="align_horizontal_left" style={{ fontSize: 16 }} />
        </IconButton>
        <IconButton selected>
          <Icon name="align_horizontal_center" style={{ fontSize: 16 }} />
        </IconButton>
        <IconButton>
          <Icon name="align_horizontal_right" style={{ fontSize: 16 }} />
        </IconButton>
      </IconButtonGroup>
    );
  },
};
