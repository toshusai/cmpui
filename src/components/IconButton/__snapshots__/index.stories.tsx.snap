// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Actions/IconButton Basic smoke-test 1`] = `
<div style="display: flex; gap: 4px;">
  <button type="button"
          class="cmpui_icon-button__root"
          data-size="M"
  >
    <span class="material-symbols-outlined"
          data-filled="true"
          style="font-size: 20px;"
    >
      arrow_selector_tool
    </span>
  </button>
  <button type="button"
          class="cmpui_icon-button__root"
          aria-selected="true"
          data-size="M"
  >
    <span class="material-symbols-outlined"
          style="font-size: 20px;"
    >
      square
    </span>
    <svg width="6"
         height="6"
         viewbox="0 0 24 24"
         fill="none"
         xmlns="http://www.w3.org/2000/svg"
         style="position: absolute; bottom: 4px; right: 4px; transform: scaleX(-1);"
    >
      <title>
        corner
      </title>
      <path d="M0 24V0.00427246L24 24H0Z"
            fill="currentColor"
      >
      </path>
    </svg>
  </button>
  <button type="button"
          class="cmpui_icon-button__root"
          data-size="M"
  >
    <span class="material-symbols-outlined"
          style="font-size: 20px;"
    >
      title
    </span>
  </button>
</div>
`;

exports[`Components/Actions/IconButton Disabled smoke-test 1`] = `
<button type="button"
        class="cmpui_icon-button__root"
        data-size="M"
        disabled
>
  <span class="material-symbols-outlined"
        style="font-size: 20px;"
  >
    star
  </span>
</button>
`;

exports[`Components/Actions/IconButton Group smoke-test 1`] = `
<div class="cmpui_icon-button__group">
  <button type="button"
          class="cmpui_icon-button__root"
          data-size="M"
  >
    <span class="material-symbols-outlined"
          data-filled="true"
          style="font-size: 16px;"
    >
      align_horizontal_left
    </span>
  </button>
  <button type="button"
          class="cmpui_icon-button__root"
          aria-selected="true"
          data-size="M"
  >
    <span class="material-symbols-outlined"
          style="font-size: 16px;"
    >
      align_horizontal_center
    </span>
  </button>
  <button type="button"
          class="cmpui_icon-button__root"
          data-size="M"
  >
    <span class="material-symbols-outlined"
          style="font-size: 16px;"
    >
      align_horizontal_right
    </span>
  </button>
</div>
`;

exports[`Components/Actions/IconButton Selected smoke-test 1`] = `
<button type="button"
        class="cmpui_icon-button__root"
        aria-selected="true"
        data-size="M"
>
  <span class="material-symbols-outlined"
        style="font-size: 20px;"
  >
    star
  </span>
</button>
`;

exports[`Components/Actions/IconButton Small smoke-test 1`] = `
<button type="button"
        class="cmpui_icon-button__root"
        data-size="S"
>
  <span class="material-symbols-outlined"
        style="font-size: 16px;"
  >
    star
  </span>
</button>
`;
